db/seeds.rb

```diff
@@ -1,9 +1,17 @@
 # frozen_string_literal: true
 
-# This file should contain all the record creation needed to seed the database with its default values.
-# The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
-#
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+cinema_hall = CinemaHall.create([{name: 'Tokyo', capacity: 200},
```

Comment:
Here you trying to create a lot of cinema halls, but assigning all the results to one variable `cinema_hall`. It's not valid that much. Consider to start with one cinema hall where you create all the seats. 

Or you can do something like this:
```
cinema_halls_params = [
 { name: 'Tokyo', capacity: 200},
 { name: 'Yokohama', capacity: 100},
 { name: 'Osaka', capacity: 100},
 { name: 'Nagoya', capacity: 100},
 { name: 'Sapporo', capacity: 100},
 { name: 'Kobe', capacity: 50},
 { name: 'Kyoto', capacity: 50},
 { name: 'Fukuoka', capacity: 50},
 { name: 'Kawasaki', capacity: 50},
 { name: 'Hiroshima', capacity: 20}
]
                                 
cinema_halls_params.each do |cinema_hall_parameters|
  cinema_hall = CinemaHall.create(cinema_hall_parameters)
  seats = CinemaHalls::GenerateSeats.new.generate_seats(cinema_hall.capacity)
  
  seats.each do |seat|
    cinema_hall.seats.create(code: seat)
  end
end
```